### PR TITLE
Switch to kotlin build scripts

### PR DESCRIPTION
**Pull request by JFronny**
AGP and Gradle support writing build scripts in kotlin, which allows for better IDE support.
Since this project is written in kotlin, doing so just makes sense.
Additionally, I have implemented auto-incrementing version codes for the prerelease flavor.

### CODE REVIEW

**Comment by C10udburst** [comment]
[build](https://github.com/recloudstream/cloudstream/actions/runs/3259203831/jobs/5351859350#step:6:9)
No files were found with the provided path: app/build/outputs/apk/debug/*.apk. No artifacts will be uploaded.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ jobs:
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
     - name: Run Gradle
-      run: ./gradlew assembleDebug
```
Comment (C10udburst) [inline .github/workflows/pull_request.yml:18]: hm why is the pull request build a `prerelease` and not `debug`?

**Review by C10udburst — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (JFronny) [reply .github/workflows/pull_request.yml:18]: It is a debug build using the prerelease flavor and is therefore debuggable, automatically version and uses the prerelase id, which seemed like a sensible setup, but a debug build with the release flavor (which would better match the current setup) is also an option, if that is preferred.

Comment (C10udburst) [reply .github/workflows/pull_request.yml:18]: why cant it be a debug flavor as it was prior to this pr?

Comment (JFronny) [reply .github/workflows/pull_request.yml:18]: release and debug are build types, not flavors, and doing things like modifying version codes is impossible from build types, so two flavors (release: a normal cloudstream build and prerelease: autogenerated versions and the prerelease suffix) were added as a second dimension of configurations.
Presented as a table:

<table>
<tr>
	<td>
	<td>prerelease
	<td>release
<tr>
	<td>release
	<td>No debugging, autogenerated version, application suffix
	<td>No debugging, normal version, no suffix
<tr>
	<td>debug
	<td>Debugging, autogenerated version, application suffix
	<td>Debugging, normal version, no suffix
</table>

Comment (JFronny) [reply .github/workflows/pull_request.yml:18]: Technically, just giving up on either automatic version codes for prereleases or manual version codes for stable releases would remove the need for them, but its up to you to decide what option of these three you prefer

Comment (C10udburst) [reply .github/workflows/pull_request.yml:18]: all right 🙏. lgtm as is

**Review by C10udburst — APPROVED** [review]

**Comment by Blatzar** [comment]
@LagradOst :whip:

**Review by LagradOst — APPROVED** [review]

**Pull request merged**